sectionproperties/pre/sections.py

```diff
@@ -293,224 +622,757 @@ def plot_geometry(self, ax=None, pause=True, labels=False, perimeter=False):
+            compound = geometry_2.align_center(geometry_1).align_top(geometry_1) + geometry_1
+            new_compound = compound.rotate_section(angle=-30)
         """
+        geoms_acc = []
+        for geom in self.geoms:
+            geoms_acc.append(geom.rotate_section(angle, rot_point, use_radians))
+        new_geom = CompoundGeometry(geoms_acc)
+        return new_geom
 
-        # check to see if there are any facets in the perimeter variable
-        if len(self.perimeter) == 0:
-            return 0
+    def mirror_section(
+        self, axis="x", mirror_point: Union[List[float], str] = "center"
+    ):
+        """Mirrors the geometry about a point on either the x or y-axis. 
 
-        # initialise perimeter variable
-        perimeter = 0
+        :param string axis: Axis about which to mirror the geometry, *'x'* or *'y'*
+        :param mirror_point: Point about which to mirror the geometry *(x, y)*. 
+            If no point is provided, mirrors the geometry about the centroid of the shape's bounding box.
+            Default = 'center'.
+        :type mirror_point: Union[list[float, float], str]
 
-        # loop through all the facets along the perimeter
-        for facet_idx in self.perimeter:
-            perimeter += self.calculate_facet_length(self.facets[facet_idx])
+        :return: Geometry object mirrored on 'axis' about 'mirror_point'
+        :rtype: :class:`sections.pre.sections.Geometry`
 
-        return perimeter
+        The following example mirrors a 200PFC section about the y-axis and the point (0, 0)::
 
+            import sectionproperties.pre.sections as sections
 
-class CustomSection(Geometry):
-    """Constructs a cross-section from a list of points, facets, holes and a user specified control
-    point.
-
-    :param points: List of points *(x, y)* defining the vertices of the cross-section
-    :type points: list[list[float, float]]
-    :param facets: List of point index pairs *(p1, p2)* defining the edges of the cross-section
-    :type facets: list[list[int, int]]
-    :param holes: List of points *(x, y)* defining the locations of holes within the cross-section.
-        If there are no holes, provide an empty list [].
-    :type holes: list[list[float, float]]
-    :param control_points: A list of points *(x, y)* that define different regions of the
-        cross-section. A control point is an arbitrary point within a region enclosed by facets.
-    :type control_points: list[list[float, float]]
-    :param shift: Vector that shifts the cross-section by *(x, y)*
-    :type shift: list[float, float]
-    :param perimeter: List of facet indices defining the perimeter of the cross-section
-    :vartype perimeter: list[int]
+            geometry_1 = sections.i_section(d=203, b=133, t_f=7.8, t_w=5.8, r=8.9, n_r=8)
+            geometry_2 = sections.rectangular_section(d=20, b=133)
+            compound = geometry_2.align_center(geometry_1).align_top(geometry_1) + geometry_1
```

Comment:
@connorferster, there is no longer a align_top method. Think example needs to be updated 👍